### PR TITLE
feat(api): adding reset password system

### PR DESCRIPTION
**Pull request by AnthoniMarie**

### CODE REVIEW

File: microservices/api/package.json [hunk 1]
```diff
@@ -47,6 +47,8 @@
   },
   "dependencies": {
     "@prisma/client": "^3.14.0",
+    "@sendgrid/mail": "^7.7.0",
+    "@types/uuid": "^8.3.4",
```
Comment (MrZalTy) [inline microservices/api/package.json:51]: @types/uuid must be in devDependencies.

File: microservices/api/src/modules/user/controllers/userController.ts [hunk 2]
```diff
@@ -33,15 +37,15 @@ async function getUsers(req: Request, res: Response): Promise<void> {
 }
 
 async function updateUser(req: Request, res: Response): Promise<void> {
-    const payload: UserUpdateDTO = req.body;
```
Comment (MrZalTy) [inline microservices/api/src/modules/user/controllers/userController.ts:36]: Prettier please :) 4 spaces not 2 spaces

File: microservices/api/src/modules/user/controllers/userController.ts [hunk 3]
```diff
@@ -50,10 +54,77 @@ async function deleteUser(req: Request, res: Response): Promise<void> {
...
+	const retrievedUser = await UserService.findByResetToken(resetToken);
+	if (!retrievedUser) {
+		throw new APIException(StatusCodes.UNAUTHORIZED, 'The password reset token is invalid or expired.');
+	}
+	const userId = retrievedUser.id;
+	await UserService.updateById(userId, {
+		password: await hashPassword(payload.password),
+		resetPasswordToken: '',
+	});
+	res.status(StatusCodes.OK);
```
Comment (MrZalTy) [inline microservices/api/src/modules/user/controllers/userController.ts:118]: res.sendStatus(StatusCodes.OK);

**Review by MrZalTy — CHANGES_REQUESTED** [review]
Some changes is required.

File: microservices/api/package.json (hunk 1 above)
Comment (AnthoniMarie) [reply microservices/api/package.json]: done

File: microservices/api/src/modules/user/controllers/userController.ts (hunk 2 above)
Comment (AnthoniMarie) [reply microservices/api/src/modules/user/controllers/userController.ts:36]: done

File: microservices/api/src/modules/user/controllers/userController.ts (hunk 3 above)
Comment (AnthoniMarie) [reply microservices/api/src/modules/user/controllers/userController.ts:118]: done

**Review by MrZalTy — APPROVED** [review]
OK

**Pull request merged**